docs/source/scheduler.rst

```diff
@@ -0,0 +1,263 @@
+Scheduler
+=========
+
+
+Overview
+--------
+
+The Treadmill scheduler is responsible for finding application placement on the
+Treadmill cell. Given the collection of servers and their attributes, such as
+available capacity and other constraints, Treadmill scheduler will place
+applications to satisfy application demands as described in its Treadmill
+manifest.
+
+Treadmill scheduler has the following high level features:
+
+ * It brokers placement decision between independent users. Treadmill scheduler
+   is multi-tenant, and different users do not coordinate their application
+   demand.
+ * Supports capacity reservations and opportunistic (non-reserved) placement.
+ * Server presence aware, will reschedule applications if the server goes down.
+ * Server topology aware, can spread or stack applications across different
+   servers, racks and pods.
+ * Support placement constraints, reboot aware.
+
+
+.. _section-paritions:
+
+Partitions
+----------
+
+Treadmill scheduler operates on a cell. The cell is collection of servers.
+Servers in the cell are divided into partitions, which are non-overlapping
+subsets of servers in the cell. Servers within a partition do not need to be
+homogenous, but they are considered interchangeable with regard to placement.
+
+The number of servers in partition can vary from single digits to thousands.
+
+When users submit application to run on a Treadmill cell, they specify the
+target partition. Partitions form a unit of administration. They can be
+restricted for subset of users, different partitions can have different alert
+thresholds, reboot schedules, or be comprised from servers that have different
+attributes that are not expressed in Treadmill manifest (such as a set of
+firewall rules, connectivity so downstream resources, very particular hardware,
+etc.)
+
+While there is no limitation on number of partitions, dividing cell into
```

Comment:
This is confusing. You first mention Treadmill operates on a partition, and that when deploying to a cell you need to specify the target partition, and then you say dividing into partitions is an anti-pattern.  Are you saying that dividing into MANY partitions is an anti pattern? What is the recommended number of partitions? What is an example of multiple partitions? Can partitions be dynamically resized while TM is running? I think this could use some more clarity.

Reply:
Rewrote the section